crates/object_detection/src/pose_detection.rs

```diff
@@ -152,11 +152,12 @@ impl PoseDetection {
         }
 
         let mut infer_request = self.network.create_infer_request()?;
-        let tensor = Tensor::new_from_host_ptr(
-            ElementType::F32,
-            &self.network.get_input()?.get_shape()?,
-            self.scratchpad.as_bytes(),
-        )?;
+        let mut tensor = Tensor::new(ElementType::F32, &self.network.get_input()?.get_shape()?)?;
```

Comment:
remove `scratchpad` from node state and pass the tensor buffer to the `load_into_scratchpad` directly. This way we save ourselves a whole copy of the image

Reply:
Done, let me know if I should also remove the `AsBytes` trait as well.
I didn't previously notice why you used the scratchpad, it all makes sense :) .

In summary, we saved the copy but still have the memory allocation in each cycle.